cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceService.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.cloudbreak.cloud.aws.connector.resource;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.services.ec2.model.DescribeInstanceTypesRequest;
+import com.amazonaws.services.ec2.model.DescribeInstanceTypesResult;
+import com.amazonaws.services.ec2.model.DiskInfo;
+import com.amazonaws.services.ec2.model.Filter;
+import com.amazonaws.services.ec2.model.InstanceStorageInfo;
+import com.google.api.client.util.Lists;
+import com.sequenceiq.cloudbreak.cloud.aws.AwsCloudFormationClient;
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonEc2Client;
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AwsCredentialView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.instance.AwsInstaceStorageInfo;
+
+@Service
+public class AwsResourceService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsResourceService.class);
+
+    @Inject
+    private AwsCloudFormationClient awsClient;
+
+    public List<AwsInstaceStorageInfo> getInstanceTypeEphemeralInfo(AuthenticatedContext authenticatedContext, List<String> instanceTypes) {
+        AwsCredentialView credentialView = new AwsCredentialView(authenticatedContext.getCloudCredential());
+        AmazonEc2Client amazonEC2Client = awsClient.createEc2Client(credentialView,
+                authenticatedContext.getCloudContext().getLocation().getRegion().value());
+        LOGGER.debug("Getting instance information on AWS for instance types: {}", instanceTypes);
+        DescribeInstanceTypesResult instanceTypesResult =  amazonEC2Client.describeInstanceTypes(new DescribeInstanceTypesRequest()
+                .withFilters(new Filter().withName("instance-type").withValues(instanceTypes)).withInstanceTypes(instanceTypes));
```

Comment:
The same instanceType is going to be sent in multiple times over ... does the API (or more specifically the com.amazonaws.services.ec2 package equivalent) handle this?

Reply:
this is the same package that is being used everywhere, it should handle the load. do you want me to cache this somewhere?

Reply:
The instance-type is the same within a hostgroup.

Reply:
this will be called only once per host group